src/hardware/vga_crtc.cpp

```diff
@@ -178,14 +178,55 @@ void vga_write_p3d5(io_port_t, io_val_t value, io_width_t)
 		// 		Register (3C0h index 13h).
 		break;
 
-	case 0x09:
+	case 0x09: {
 		if (IS_VGA_ARCH) {
 			vga.config.line_compare = (vga.config.line_compare & 0x5ff) |
 			                          (val & 0x40) << 3;
 		}
+
+		const auto old_val = vga.crtc.maximum_scan_line;
+		const auto new_val = MaximumScanLineRegister{val};
+
 		vga.crtc.maximum_scan_line.data = val;
-		VGA_StartResize();
-		break;
+
+		if (vga.draw.is_double_scanning) {
+			auto flipped = new_val;
+			flipped.start_vertical_blanking_bit9.flip();
+
+			// Only true if only `start_vertical_blanking_bit9` has
+			// been changed and all other bits are unchanged.
+			if (old_val.data == flipped.data) {
+				VGA_StartResize();
+			}
+		} else {
+			auto flipped = new_val;
+			flipped.maximum_scan_line = new_val.maximum_scan_line ^ 1;
+			flipped.start_vertical_blanking_bit9.flip();
+			flipped.is_scan_doubling_enabled.flip();
+
+			// Only true if ALL of the following three bits have
+			// been changed:
+			//   - LSB of `maximum_scan_line`
+			//   - `start_vertical_blanking_bit9`
+			//   - `is_scan_doubling_enabled`
+			// and all other bits are unchanged.
+			//
+			// This revised criteria was taken from DOSBox X. The
+			// old DOSBox criteria was "if ALL bits have been
+			// changed EXCEPT for `line_compare_bit9`"; I can't see
+			// how could that possibly work.
+			//
+			if (old_val.data == flipped.data) {
```

Comment:
Similarly, I think this is easy to understand if you read the comment. The alternative way via bit-twiddling is rather obscure (took me a good ten minutes to decipher it as I'm out of practice with that stuff 😎).

Reply:
So much easier to read and understand. Doesn't get easier than this!

Reply:
Thanks man 😎